src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationServiceImpl.java

```diff
@@ -301,8 +308,14 @@ private Negotiation findEntityById(String negotiationId, boolean includeDetails)
    * @param includeDetails whether the negotiation returned include details
    * @return the Negotiation with specified negotiationId
    */
+  @Override
   public NegotiationDTO findById(String negotiationId, boolean includeDetails)
       throws EntityNotFoundException {
+    Long userID = AuthenticatedUserContext.getCurrentlyAuthenticatedUserInternalId();
```

Comment:
you can move this statement after the following `if`, so it is not called if unnecessary (really pedantic :) )